### ISSUE TITLE
[Bug][Master] Transform command  to WorkflowExecutionRunnable failed due to t_ds_trigger_relation duplicate data

### ISSUE THREAD

**Issue by peak-xu** [apache/dolphinscheduler#15864]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

can't start workflow

### What you expected to happen

workflow is running ~~~

### How to reproduce

edit workflow and save online and run ,you will see that the workflow cannot run,and master log have this error 

### Version

3.2.x

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by peak-xu** [comment]
<img width="1196" alt="image" src="https://github.com/apache/dolphinscheduler/assets/49389493/fdd9d81b-4cbb-44d8-a6a7-0afa292b1f31">

**Comment by peak-xu** [comment]
Miraculously, he will be normal again after a while, I don't know why

**Comment by ruanwenjun** [comment]
Could you please find out which interface throw this exception.

**Comment by peak-xu** [comment]
> Could you please find out which interface throw this exception.

<img width="1204" alt="image" src="https://github.com/apache/dolphinscheduler/assets/49389493/8855e498-5bc2-4e72-804d-487a6a09b238">

**Comment by ruanwenjun** [comment]
It seems related to trigger relation,  Is easy to reproduce this problem?

**Comment by ruanwenjun** [comment]
> > Could you please find out which interface throw this exception.
> 
> <img alt="image" width="1204" src="https://private-user-images.githubusercontent.com/49389493/323431593-8855e498-5bc2-4e72-804d-487a6a09b238.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0MTkwOTEsIm5iZiI6MTcxMzQxODc5MSwicGF0aCI6Ii80OTM4OTQ5My8zMjM0MzE1OTMtODg1NWU0OTgtNWJjMi00ZTcyLTgwNGQtNDg3YTZhMDliMjM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE4VDA1Mzk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkMjE5YTIzZGJkN2M2MTIyMWU2MTEwZmRkNjk3MjQwNDdmOTIzMDhiZTkzNzFhYzUxY2YzMDExODZmYjFiNTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DBysn1EhmrotXRWimdOpeYw3dyad-XhuiwcdZCK-_Kk">

Could you please find out the duplicate data from db.
```
select * from t_ds_trigger_relation WHERE trigger_type = 2 and job_id =  xx;
```
Is there exist job_id is null?

**Comment by peak-xu** [comment]
> select * from t_ds_trigger_relation WHERE trigger_type = 2 and job_id =  xx;

<img width="1023" alt="image" src="https://github.com/apache/dolphinscheduler/assets/49389493/24a809dc-1c0d-412c-bdd7-902a1274da8c">
No null value exists
It may have something to do with the way it is used. If so, it is really not easy to reproduce.

**Comment by peak-xu** [comment]
select `trigger_type`,`job_id`,`trigger_code`,count(id) as cnt from t_ds_trigger_relation group by `trigger_type`,`job_id`,`trigger_code` having cnt>1;
Duplicate data exists ! this table uk is  `t_ds_trigger_relation_UN` (`trigger_type`,`job_id`,`trigger_code`)

**Comment by ruanwenjun** [comment]
> select trigger_type,job_id,trigger_code,count(id) as cnt from t_ds_trigger_relation group by trigger_type,job_id,trigger_code having cnt>1;
Duplicate data exists ! this table uk is  t_ds_trigger_relation_UN (trigger_type,job_id,trigger_code)

It's strange, the uk doesn't work?

**Comment by peak-xu** [comment]
> > select trigger_type,job_id,trigger_code,count(id) as cnt from t_ds_trigger_relation group by trigger_type,job_id,trigger_code having cnt>1;
> > Duplicate data exists ! this table uk is  t_ds_trigger_relation_UN (trigger_type,job_id,trigger_code)
> 
> It's strange, the uk doesn't work?

sorry,I didn't describe it clearly.

select * from t_ds_trigger_relation WHERE trigger_type = 2 and job_id =  xx;

The query in the source code only has two fields, so the result is definitely not unique

**Comment by ruanwenjun** [comment]
@peak-xu Could you please find out how to reproduce this problem, seems serious.

**Comment by ruanwenjun** [comment]
Anyway the data in t_ds_trigger_relation is not unique by type and jobId, so I submit https://github.com/apache/dolphinscheduler/pull/15883 to fix  this problem.

**Comment by peak-xu** [comment]
> @peak-xu Could you please find out how to reproduce this problem, seems serious.

I cant to reproduce,I don't know what caused it

**Comment by XiaoZhiSnow** [comment]
reboot mysql which ds used , you will reproduce it.

**Comment by peak-xu** [comment]
> reboot mysql which ds used , you will reproduce it.

It is possible that we have restarted MySQL,but I don't know why ds needs to write it again

**Comment by ruanwenjun** [comment]
This might caused by `@Transaction` didn't work? I am not sure.

**Comment by hogger2099** [comment]
This problem has also occurred to me.
select * from t_ds_trigger_relation WHERE trigger_type = 2 and job_id = xx;
job_id is not null
Our DS was running steadily for 2 months, and then suddenly it happened.

**Comment by zlsdmx** [comment]
> reboot mysql which ds used , you will reproduce it.
Yes, mine also happened under this situation.
Moreover, the complement function cannot run anymore。 the original ordinary scheduling process is not affected.

**Comment by zlsdmx** [comment]
> > reboot mysql which ds used , you will reproduce it.
>  Yes, mine also happened under this situation.
Moreover, the complement function cannot run anymore。 the original ordinary scheduling process is not affected.

**Comment by zlsdmx** [comment]
By comparing the source codes of different versions 3.2.0~3.2.2, I found that this problem has been fixed in the 3.2.2-release version. And added this issue address in the code:
`org.apache.dolphinscheduler.service.process.TriggerRelationServiceImpl#saveCommandTrigger#row-70`
![image](https://github.com/user-attachments/assets/ef41c9d7-9e87-4766-bf74-d62cd18064ec)
